libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -207,6 +207,7 @@ class YulUtilFunctions
 	std::string mappingIndexAccessFunction(MappingType const& _mappingType, Type const& _keyType);
 
 	/// @returns a function that reads a value type from storage.
```

Comment:
The documentation here says this function is only meant to be used for value types.

Reply:
Would it be more complicated to use special functions for non-value types? I think the separation makes sense because non-value types are much more complicated and need to allocate.

Reply:
Totally makes sense, will introduce another function for non-value types

Reply:
I added two functions which are called via old function depending on if type is value or reference.